packages/grafana-ui/src/components/Link/TextLink.tsx

```diff
@@ -0,0 +1,107 @@
+import { css } from '@emotion/css';
+import React, { AnchorHTMLAttributes, forwardRef } from 'react';
+
+import { GrafanaTheme2, locationUtil, textUtil, ThemeTypographyVariantTypes } from '@grafana/data';
+
+import { useTheme2 } from '../../themes';
+import { IconName, IconSize } from '../../types';
+import { Icon } from '../Icon/Icon';
+import { customWeight } from '../Text/utils';
+
+import { Link } from './Link';
+
+interface TextLinkProps extends Omit<AnchorHTMLAttributes<HTMLAnchorElement>, 'target' | 'rel'> {
+  /** url to which redirect the user, external or internal */
+  href: string;
+  /** Color to use for text */
+  color?: keyof GrafanaTheme2['colors']['text'];
+  /** Specify if the link will redirect users to a page in or out Grafana */
+  external?: boolean;
+  /** True if the element should be displayed inline with surrounding text, false if it should be displayed as a block */
+  inline?: boolean;
+  /** The default variant is 'body'. To fit another styles set the correspondent variant as it is necessary also to adjust the icon size */
+  variant?: keyof ThemeTypographyVariantTypes;
+  /** Override the default weight for the used variant */
+  weight?: 'light' | 'regular' | 'medium' | 'bold';
+  /** Set the icon to be shown. An external link will show the 'external-link-alt' icon as default.*/
+  icon?: IconName;
+  children: string;
+}
+
+const svgSizes: {
+  [key in keyof ThemeTypographyVariantTypes]: IconSize;
+} = {
+  h1: 'xl',
+  h2: 'xl',
+  h3: 'lg',
+  h4: 'lg',
+  h5: 'md',
+  h6: 'md',
+  body: 'md',
+  bodySmall: 'xs',
+};
+
+export const TextLink = forwardRef<HTMLAnchorElement, TextLinkProps>(
+  ({ href, color, external = false, inline, variant = 'body', weight, icon, children, ...rest }, ref) => {
```

Comment:
Maybe you can set `inline=true` here instead of in `getLinkStyles`, since it essentially defaults to true anyways. It makes it a bit more readable.